fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -226,36 +238,100 @@ private void registerMaterializedViews(OptExpression logicOperatorTree, ConnectC
-            materializationContext.setScanMvOutputExpressions(scanMvPlans.second.getOutputColumn());
+            materializationContext.setOutputMapping(outputMapping);
             context.addCandidateMvs(materializationContext);
         }
     }
 
+    private LogicalOlapScanOperator createScanMvExpression(MaterializationContext materializationContext) {
+        MaterializedView mv = materializationContext.getMv();
+
+        ImmutableMap.Builder<ColumnRefOperator, Column> colRefToColumnMetaMapBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<Column, ColumnRefOperator> columnMetaToColRefMapBuilder = ImmutableMap.builder();
+        ImmutableList.Builder<ColumnRefOperator> outputVariablesBuilder = ImmutableList.builder();
+
+        ColumnRefFactory columnRefFactory = materializationContext.getQueryRefFactory();
+        int relationId = columnRefFactory.getNextRelationId();
+        for (Column column : mv.getFullSchema()) {
+            ColumnRefOperator columnRef = columnRefFactory.create(column.getName(),
+                    column.getType(),
+                    column.isAllowNull());
+            columnRefFactory.updateColumnToRelationIds(columnRef.getId(), relationId);
+            columnRefFactory.updateColumnRefToColumns(columnRef, column, mv);
+            outputVariablesBuilder.add(columnRef);
+            colRefToColumnMetaMapBuilder.put(columnRef, column);
+            columnMetaToColRefMapBuilder.put(column, columnRef);
+        }
+
+        Map<Column, ColumnRefOperator> columnMetaToColRefMap = columnMetaToColRefMapBuilder.build();
+        DistributionInfo distributionInfo = mv.getDefaultDistributionInfo();
+        Preconditions.checkState(distributionInfo instanceof HashDistributionInfo);
+        HashDistributionInfo hashDistributionInfo = (HashDistributionInfo) distributionInfo;
+        List<Column> distributedColumns = hashDistributionInfo.getDistributionColumns();
+        List<Integer> hashDistributeColumns = new ArrayList<>();
+        for (Column distributedColumn : distributedColumns) {
+            hashDistributeColumns.add(columnMetaToColRefMap.get(distributedColumn).getId());
+        }
+
+        HashDistributionDesc hashDistributionDesc =
+                new HashDistributionDesc(hashDistributeColumns, HashDistributionDesc.SourceType.LOCAL);
+
+        List<Long> selectPartitionIds = Lists.newArrayList();
+        Set<String> excludedPartitions = mv.getPartitionNamesToRefreshForMv();
+        List<String> selectedPartitionNames = mv.getPartitionNames()
+                .stream().filter(name -> !excludedPartitions.contains(name)).collect(Collectors.toList());
+        for (Partition p : mv.getPartitions()) {
+            if (selectedPartitionNames.contains(p.getName())) {
+                selectPartitionIds.add(p.getId());
+            }
+        }
+        PartitionNames partitionNames = new PartitionNames(false, selectedPartitionNames);
```

Comment:
Mv scan with **not**  out of date partition, it still need use base table partition range as predicate?

Reply:
no, it is not necessary to add partition range predicate by using the method.